x/launch/keeper/request.go

```diff
@@ -133,6 +133,40 @@ func ApplyRequest(
 	k Keeper,
 	launchID uint64,
 	request types.Request,
+) error {
+	if err := CheckRequest(ctx, k, launchID, request); err != nil {
+		return err
+	}
+
+	switch requestContent := request.Content.Content.(type) {
+	case *types.RequestContent_GenesisAccount:
+		ga := requestContent.GenesisAccount
+		k.SetGenesisAccount(ctx, *ga)
+	case *types.RequestContent_VestingAccount:
+		va := requestContent.VestingAccount
+		k.SetVestingAccount(ctx, *va)
+	case *types.RequestContent_AccountRemoval:
+		ar := requestContent.AccountRemoval
+		k.RemoveGenesisAccount(ctx, launchID, ar.Address)
+		k.RemoveVestingAccount(ctx, launchID, ar.Address)
+	case *types.RequestContent_GenesisValidator:
+		ga := requestContent.GenesisValidator
+		k.SetGenesisValidator(ctx, *ga)
+	case *types.RequestContent_ValidatorRemoval:
+		vr := requestContent.ValidatorRemoval
+		k.RemoveGenesisValidator(ctx, launchID, vr.ValAddress)
+	default:
+		return spnerrors.Critical("unknown request content type")
+	}
+	return nil
+}
+
+// CheckRequest verifies that a request can be applied
+func CheckRequest(
```

Comment:
We should create the `TestCheckRequest` test to test request checking

`TestApplyRequest` would keep the state transition assertion and should verify that it fails if a request is invalid (without testing all invalid requests)